src/runtime-utils/mount.ts

```diff
@@ -59,15 +59,63 @@ export async function mountSuspended<T>(
   const vueApp = tryUseNuxtApp()?.vueApp
     // @ts-expect-error untyped global __unctx__
     || globalThis.__unctx__.get('nuxt-app').tryUse().vueApp
-  const { render, setup, data, computed, methods } = component as DefineComponent<Record<string, unknown>, Record<string, unknown>>
+  const {
+    computed,
+    data,
+    methods,
+    render,
+    setup,
+  } = component as DefineComponent<Record<string, unknown>, Record<string, unknown>>
 
   let setupContext: SetupContext
   let setupState: Record<string, unknown>
   const setProps = reactive<Record<string, unknown>>({})
 
+  let interceptedEmit: ((event: string, ...args: unknown[]) => void) | null = null
+  /**
+   * Intercept the emit for testing purposes.
+   *
+   * @remarks
+   * Using this function ensures that the emit is not intercepted multiple times
+   * and doesn't duplicate events.
+   *
+   * @param emit - The original emit from the component's context.
+   *
+   * @returns An intercepted emit that will both emit from the component itself
+   * and from the top level wrapper for assertions via
+   * {@link import('@vue/test-utils').VueWrapper.emitted()}.
+   */
+  function getInterceptedEmitFunction(
+    emit: ((event: string, ...args: unknown[]) => void),
+  ): ((event: string, ...args: unknown[]) => void) {
+    if (emit !== interceptedEmit) {
+      interceptedEmit = interceptedEmit ?? ((event, ...args) => {
+        emit(event, ...args)
+        setupContext.emit(event, ...args)
+      })
+    }
+
+    return interceptedEmit
+  }
+
+  /**
+   * Intercept emit for assertions in populate wrapper emitted.
+   */
+  function interceptEmitOnCurrentInstance(): void {
+    const currentInstance = getCurrentInstance()
+    if (currentInstance == null) {
+      return
+    }
+
+    currentInstance.emit = getInterceptedEmitFunction(currentInstance.emit)
+  }
```

Comment:
I think we also need to apply this fix in `render.ts` (and maybe we could consider abstracting the code in a future PR so we don't just have two copies of the same logic)